beamer/agent/state_machine.py

```diff
@@ -163,9 +161,8 @@ def _timestamp_is_l1_finalized(
     # FIXME: Remove this for the above once the contracts with the event are deployed
     target_chain_finality = context.finality_periods.get(target_chain_id)
     if target_chain_finality is None:
-        target_chain_finality = context.request_manager.functions.finalityPeriods(
-            target_chain_id
-        ).call()
+        target_chain = context.request_manager.functions.chains(target_chain_id).call()
```

Comment:
This looks like an inefficiency in the old code, but since we store and update the finality period in memory, we can reuse `context.finality_periods[target_chain_id]` instead of fetching it with another network call here